tests/src/astro/electromagnetism/unitTestYarkovskyAcceleration.cpp

```diff
@@ -0,0 +1,207 @@
+    };
+
+    TUDAT_CHECK_MATRIX_CLOSE( computedYarkovskyAcceleration, expectedYarkovskyAcceleration, 1.e-10 );
+}
+
+//! Test the implementation of the Yarkovsky Acceleration Model
+BOOST_AUTO_TEST_CASE( testYarkovskyAccelerationSimple )
+{
+    const Eigen::Vector6d state = { 2.0 * AU, AU, 1.0e10, 3.0e5, 4.0e5, 12.0e5
+    };
+
+    const Eigen::Vector3d expectedYarkovskyAcceleration =
+            yarkovskyParameter * 0.19982142476807888 * Eigen::Vector3d { 3.0, 4.0, 12.0 } / 13.0;
+    const Eigen::Vector3d computedYarkovskyAcceleration = electromagnetism::computeYarkovskyAcceleration(
+            yarkovskyParameter,
+            state );
+
+    TUDAT_CHECK_MATRIX_CLOSE( computedYarkovskyAcceleration, expectedYarkovskyAcceleration, 1.0e-10 );
+}
+
+Eigen::Vector6d bodyState = Eigen::Vector6d { AU, 0, 0, 12.0e5, 3.0e5, 4.0e5 };
+
+Eigen::Vector6d getBodyState( ) { return bodyState; }
+
+Eigen::Vector6d centralBodyState = Eigen::Vector6d::Zero( );
+
+Eigen::Vector6d getCentralBodyState( ) { return centralBodyState; }
+
+//! Test the class implementation of the Yarkovsky Acceleration model (updateMembers)
+BOOST_AUTO_TEST_CASE( testYarkovskyAccelerationModelClassUpdateMembers )
+{
+    electromagnetism::YarkovskyAccelerationPointer yarkovskyAccelerationModel = std::make_shared< electromagnetism::YarkovskyAcceleration >(
+            yarkovskyParameter,
+            &getBodyState,
+            &getCentralBodyState );
+
+    yarkovskyAccelerationModel->updateMembers( 0.0 );
+    const Eigen::Vector3d expectedYarkovskyAcceleration =
+            yarkovskyParameter * Eigen::Vector3d { 12.0, 3.0, 4.0 } / 13.0;
+    const Eigen::Vector3d computedYarkovskyAcceleration = yarkovskyAccelerationModel->getAcceleration( );
+
+    TUDAT_CHECK_MATRIX_CLOSE( computedYarkovskyAcceleration, expectedYarkovskyAcceleration, 1.0e-10 )
+}
+
+//! Test the complete implementation of the yarkovsky Acceleration using the drift in semi-major axis for a circular orbit
+//! Reference: Pérez-Hernández, J. A., & Benet, L. (2022). Non-zero Yarkovsky acceleration for near-Earth asteroid (99942) Apophis. Communications Earth & Environment, 3(1), Article 1. https://doi.org/10.1038/s43247-021-00337-x
+//! For Circulaar Orbit
+BOOST_AUTO_TEST_CASE( testYarkovskyAccelerationCircular )
+{
+    using namespace simulation_setup;
+
+    spice_interface::loadStandardSpiceKernels( );
+
+    // Simulation Setup Constants
+    const std::string frameOrigin = "SSB";
+    const std::string frameOrientation = "ECLIPJ2000";
+    const double simulationStartEpoch = 10.0 * physical_constants::JULIAN_DAY;
+    const double simulationDuration = 1000.0 * physical_constants::JULIAN_DAY;
+    const double simulationEndEpoch = simulationStartEpoch + simulationDuration;
+    const double fixedStepSize = 1000.0;
+
+    // Initial Conditions
+    const double initialSemiMajorAxis = 0.75 * AU;
+    const double initialEccentricity = 0.0;
+    const double initialInclination = 0.0;
```

Comment:
Why not add a small inclination/eccentricity? You're using Kepler elements in your check, and your initial definition is at a singularity. This should not matter for the definition of the semi-major axis, but it's still a little risky. Also, by setting e=0, this test is missing the distance-dependency that the model should have

Reply:
Indeed. Because the acceleration is so small, I chose $e=0$, so that I could measure the drift in $a$ over a longer time period. I didn't think of saving $a$ as a dependent variable but will do that and then also include an eccentricity.

Reply:
Thanks! What you can also do is make the value of A (yarkovskyParameter) much bigger (say by a factor 100). None of the governing equations depend on a certain range for A, the main reason it has its value as it does it to make it representative of a realistic case. But, the validity of the code is not contingent upon having a realistic value for A. One risk of making A *too* large is that you may find that the semi-major axis drift is no longer linear (since da/dt depends on a, having da/dt very large will make the behaviour non-linear)